hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/AWSClientConfig.java

```diff
@@ -407,6 +413,48 @@ private static void initSigner(Configuration conf,
     }
   }
 
+  /**
+   *
+   * @param conf hadoop configuration
+   * @param clientConfig client configuration to update
+   * @param awsServiceIdentifier service name
+   */
+  private static void initRequestHeaders(Configuration conf,
+      ClientOverrideConfiguration.Builder clientConfig, String awsServiceIdentifier) {
+    String configKey = null;
+    switch (awsServiceIdentifier) {
+      case AWS_SERVICE_IDENTIFIER_S3:
+        configKey = CUSTOM_HEADERS_S3;
+        break;
+      case AWS_SERVICE_IDENTIFIER_STS:
+        configKey = CUSTOM_HEADERS_STS;
+        break;
+      default:
+        // Nothing to do. The original signer override is already setup
+    }
+    if (configKey != null) {
+      String[] customHeaders = conf.getTrimmedStrings(configKey);
+      if (customHeaders == null || customHeaders.length == 0) {
+        LOG.debug("No custom headers specified");
+        return;
+      }
+
+      for (String customHeader : customHeaders) {
```

Comment:
HADOOP-18980/#6406 addes split and parse of headers

Reply:
Unknown headers are neglected

Reply:
Thank you, I have revised the PR based on the feedback.
>How do the s3 and STS stores react to unknown headers? What do they support today? I'm curious

They ignore unknown headers. there are multiple use cases as mentioned [here](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonRequestHeaders.html). Most common I could think of is `Content-Type` which is required in write requests with body being html or something.